x/authority/keeper/msg_server_remove_chain_info.go

```diff
@@ -0,0 +1,45 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/node/x/authority/types"
+)
+
+// RemoveChainInfo removes the chain info for the specified chain id
+func (k msgServer) RemoveChainInfo(
+	goCtx context.Context,
+	msg *types.MsgRemoveChainInfo,
+) (*types.MsgRemoveChainInfoResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check the authorization for this message against the authorization list
+	err := k.CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, errors.Wrap(types.ErrUnauthorized, err.Error())
+	}
+
+	chainInfo, found := k.GetChainInfo(ctx)
+	if !found {
+		return nil, types.ErrChainInfoNotFound
+	}
+
+	updatedChainInfo := RemoveChain(chainInfo, msg.ChainId)
+	k.SetChainInfo(ctx, updatedChainInfo)
+	return &types.MsgRemoveChainInfoResponse{}, nil
+}
+
+// RemoveChain removes the chain with the given chainID from the chainInfo
+// If the chain does not exist, the chainInfo is returned as is
+func RemoveChain(chainInfo types.ChainInfo, chainID int64) types.ChainInfo {
```

Comment:
```suggestion
func removeChain(chainInfo types.ChainInfo, chainID int64) types.ChainInfo {
```
Not sure we specifically need a sub function here as the logic is pretty small

Reply:
This is a bit easier to test. I'd like to keep it